src/Veldrid/MTL/MTLGraphicsDevice.cs

```diff
@@ -423,12 +403,8 @@ private protected override void UpdateBufferCore(DeviceBuffer buffer, uint buffe
 
         private protected override void WaitForIdleCore()
         {
-            MTLCommandBuffer lastCB = default(MTLCommandBuffer);
-            lock (_submittedCommandsLock)
-            {
-                lastCB = _latestSubmittedCB;
-                ObjectiveCRuntime.retain(lastCB.NativePtr);
-            }
+            MTLCommandBuffer lastCB = _latestSubmittedCB;
```

Comment:
This is still being manually released? Command buffers can be safely used until they're in a completed state, at which point they'll be automatically released. Therefore it should be safe to not retain or release here?

Reply:
the assumption is it's not has age because the usage is beyond commit/present. But it's moot at this point because the assumptions don't hold.